lib/compiler_rt/memcpy.zig

```diff
@@ -7,18 +7,33 @@ comptime {
 
 pub fn memcpy(noalias dest: ?[*]u8, noalias src: ?[*]const u8, len: usize) callconv(.C) ?[*]u8 {
     @setRuntimeSafety(false);
+    if (len == 0) return dest;
 
-    if (len != 0) {
-        var d = dest.?;
-        var s = src.?;
-        var n = len;
-        while (true) {
-            d[0] = s[0];
-            n -= 1;
-            if (n == 0) break;
-            d += 1;
-            s += 1;
-        }
+    var d = dest.?;
+    var s = src.?;
+    const wsize = @sizeOf(usize);
+    const off = len % wsize;
+
+    // align comparison to usize
+    var n = off;
+    while (true) {
+        if (n == 0) break;
+        d[0] = s[0];
+        n -= 1;
+        d += 1;
+        s += 1;
+    }
+
+    // compare whole words at a time rather than single bytes
+    var dw = @ptrCast([*]usize, @alignCast(@alignOf(usize), d));
```

Comment:
Is this alignCast correct?

Seems like it would depend on the alignment of `d + len`